home/.chezmoiscripts/universal/run_onchange_before_11-install-wazuh-agent.sh.tmpl

```diff
@@ -0,0 +1,141 @@
+{{- if ne .host.distro.family "windows" -}}
+#!/usr/bin/env bash
+# @file Wazuh agent installation and registration
+# @brief Ensures Wazuh Agent is installed and the agent joined to the given Wazuh Manager
+# @description
+#     This script ensures Wazuh Agent is installed and joined to Wazuh Manager under the `default` group. If Wazuh Manager IP is not provided
+#     in the input, agent installation is skipped. On MacOS and Archlinux, the script builds the agent and then installs.
+#
+#     No configuraiton other than the Manager IP is added to the config file by this script. Please refer the
+#     [Deployment variables document](https://documentation.wazuh.com/current/user-manual/deployment-variables/deployment-variables.html),
+#     relevant to the OS you are using for more information.
+#
+#     ## Configuration Variables
+#
+#     The following chart details the input variable(s) that are used to determine the configuration of the Wazuh Agent:
+#
+#     | Variable        | Description                                                |
+#     |-----------------|------------------------------------------------------------|
+#     | `wazuh.manager` | The IP address where Wazuh Manager is available            |
```

Comment:
Let's also make WAZUH_MANAGER use the secret then the environment variable and then `wazuh.{{ PUBLIC_SERVICES_ENDPOINT }}` as the value for WAZUH_MANAGER. You can update the wazuh.manager in `.chezmoi.yaml.tmpl` to be `wazuhManager` instead and populate it with this new method.

Reply:
That way, you can just use `{{ .host.wazuhManager }}` in this script.

Reply:
I have made these changes, but, do we need to store the manager as a secret? Just `wazuh.{{ PUBLIC_SERVICES_ENDPOINT }}` in the config file should be sufficient, right? This would also let anyone change the value easily without having to know template syntax. If I am right, `WAZUH_MANAGER` environment variable is only used for installing agents (on all supported platforms). So, it may not have use after the setup is complete.